unity-renderer/Assets/UIComponents/Tests/TagComponentViewTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NUnit.Framework;
+using UnityEngine;
+
+public class TagComponentViewTests
+{
+    private TagComponentView tagComponent;
+
+    [SetUp]
+    public void SetUp() { tagComponent = BaseComponentView.Create<TagComponentView>("Tag"); }
+
+    [TearDown]
+    public void TearDown()
+    {
+        tagComponent.Dispose();
+        GameObject.Destroy(tagComponent.gameObject);
+    }
+
+    [Test]
+    public void ConfigureTagCorrectly()
+    {
+        // Arrange
+        TagComponentModel testModel = new TagComponentModel
+        {
+            icon = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero),
+            text = "Test"
+        };
+
+        // Act
+        tagComponent.Configure(testModel);
+
+        // Assert
+        Assert.AreEqual(testModel, tagComponent.model, "The model does not match after configuring the button.");
+    }
+
+    [Test]
+    public void RefreshTagCorrectly()
+    {
+        // Arrange
+        Sprite testSprite = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero);
+        string testText = "Test";
+
+        tagComponent.model.icon = testSprite;
+        tagComponent.model.text = testText;
+
+        // Act
+        tagComponent.RefreshControl();
+
+        // Assert
+        Assert.AreEqual(testSprite, tagComponent.model.icon, "The icon does not match in the model.");
+        Assert.AreEqual(testText, tagComponent.model.text, "The text does not match in the model.");
+    }
+
+    [Test]
+    public void SetTagTextCorrectly()
+    {
+        // Arrange
+        string testText = "Test";
+
+        // Act
+        tagComponent.SetText(testText);
+
+        // Assert
+        Assert.AreEqual(testText, tagComponent.model.text, "The text does not match in the model.");
+        Assert.AreEqual(testText, tagComponent.text.text, "The button text does not match.");
+    }
+
+    [Test]
+    [TestCase(true)]
+    [TestCase(false)]
+    public void SetTagIconCorrectly(bool isNullIcon)
+    {
+        // Arrange
+        Sprite testSprite = isNullIcon ? null : Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero);
```

Comment:
The `Texture2D` objects have to be destroyed manually to avoid leaks.

Reply:
Good catch! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/531831692da512ed2d0095f4e2c0d08be70ee7be